tests/catch/unit/stream/hipStreamGetPriority.cc

```diff
@@ -28,46 +28,62 @@ Testcase Scenarios :
 #include <hip_test_common.hh>
 
 /**
- * Negative tests for hipStreamGetPriority api.
+ * Check the error returned when a null stream is used.
  */
-TEST_CASE("Unit_hipStreamGetPriority_Negative") {
-  hipStream_t stream = 0;
-  REQUIRE(hipStreamGetPriority(stream, nullptr) == hipErrorInvalidValue);
+TEST_CASE("Unit_hipStreamGetPriority_NullStream") {
+  int priority = 0;
+  HIP_CHECK_ERROR(hipStreamGetPriority(nullptr, &priority), hipErrorInvalidResourceHandle);
 }
 
 /**
- * Create stream with higher priority for the priority range returned.
+ * Check the error returned when an invalid pointer to a priority is used.
  */
-TEST_CASE("Unit_hipStreamGetPriority_higher") {
-  int priority_low = 0;
-  int priority_high = 0;
-  int devID = GENERATE(range(0, HipTest::getDeviceCount()));
-  HIP_CHECK(hipSetDevice(devID));
-  HIP_CHECK(hipDeviceGetStreamPriorityRange(&priority_low, &priority_high));
-  hipStream_t stream;
-  HIP_CHECK(hipStreamCreateWithPriority(&stream, hipStreamNonBlocking,
-                                                           priority_high-1));
-  int priority = 0;
-  HIP_CHECK(hipStreamGetPriority(stream, &priority));
-  REQUIRE(priority == priority_high);
+TEST_CASE("Unit_hipStreamGetPriority_InvalidPriorityPointer") {
+  hipStream_t stream{};
+  HIP_CHECK(hipStreamCreate(&stream));
+  HIP_CHECK_ERROR(hipStreamGetPriority(stream, nullptr), hipErrorInvalidValue);
   HIP_CHECK(hipStreamDestroy(stream));
 }
 
 /**
- * Create stream with lower priority for the priority range returned.
+ * Create stream and check priority.
  */
-TEST_CASE("Unit_hipStreamGetPriority_lower") {
+TEST_CASE("Unit_hipStreamGetPriority_happy") {
   int priority_low = 0;
   int priority_high = 0;
   int devID = GENERATE(range(0, HipTest::getDeviceCount()));
   HIP_CHECK(hipSetDevice(devID));
   HIP_CHECK(hipDeviceGetStreamPriorityRange(&priority_low, &priority_high));
-  hipStream_t stream;
-  HIP_CHECK(hipStreamCreateWithPriority(&stream, hipStreamNonBlocking,
-                                                           priority_low+1));
+  hipStream_t stream{};
   int priority = 0;
-  HIP_CHECK(hipStreamGetPriority(stream, &priority));
-  REQUIRE(priority_low == priority);
+  SECTION("Default Priority") {
+    HIP_CHECK(hipStreamCreate(&stream));
+    HIP_CHECK(hipStreamGetPriority(stream, &priority));
+    // valid priority
+    // Lower the value higher the priority, higher the value lower the priority
+    REQUIRE(priority_low >= priority);
+    REQUIRE(priority >= priority_high);
+  }
+  SECTION("High Priority") {
+    HIP_CHECK(hipStreamCreateWithPriority(&stream, hipStreamDefault, priority_high));
+    HIP_CHECK(hipStreamGetPriority(stream, &priority));
+    REQUIRE(priority == priority_high);
+  }
+  SECTION("Higher Priority") {
+    HIP_CHECK(hipStreamCreateWithPriority(&stream, hipStreamNonBlocking, priority_high - 1));
```

Comment:
check that priority_high isn't equal to priority_low

Reply:
I don't understand why this is relevant? This test checks that priority to clamped to the range of priorities.
If priorities aren't supported then priority_high will equal priority_low which will equal 0

Reply:
Which doesn't work because then you'll -1 to the API expecting it to be "fine"